src/university_chatbot/university_conversation_logic_adapter.py

```diff
@@ -43,30 +43,72 @@ def find_best_tags_coverage(self, documents, tags):
-                                                                           extracted_lemmas)
-            if len(docs_by_lemmas) > 0:
-                find_coverage_res = self.find_best_tags_coverage(docs_by_lemmas, extracted_lemmas)
-                if find_coverage_res is not None:
-                    result_document_lemmas, confidence_by_lemmas = find_coverage_res
-        if confidence_by_lemmas + confidence_by_tags > -2 and result_document_tags is not None:
-            if confidence_by_tags > confidence_by_lemmas:
-                res = Statement(result_document_tags)
-                res.confidence = 1
+        tags = self.create_tags(statement)
+        docs_main_collection = self.db.get_docs_from_collection_by_tags_list(Configuration.MAIN_COLLECTION.value, tags)
+        docs_phrases_collection = self.db.get_docs_from_collection_by_tags_list(Configuration.PHRASES_COLLECTION.value, tags)
+        result_main_collection = ['', 0.0]
+        result_phrases_collection = ['', 0.0]
+        if len(docs_main_collection) > 0:
+            result_main_collection = self.find_best_tags_coverage(docs_main_collection, tags)
+        if len(docs_phrases_collection) > 0:
+            result_phrases_collection = self.find_best_tags_coverage(docs_phrases_collection, tags)
+        return self.prepare_correct_answer(result_main_collection, result_phrases_collection)
+
+    def create_tags(self, statement):
+        tags = list(self.sentence_filter.extract_complex_lemmas_and_filter_stopwords(statement.text))
+        single_tags, complex_tags = self.sentence_filter.split_to_single_and_complex_lemmas(tags)
+        if len(complex_tags) != 0:
+            tags = self.create_combinated_tags_list(single_tags, complex_tags)
+        return tags
+
+    def create_combinated_tags_list(self, single_tag, complex_tag):
+        single_tags_from_complex_tags = self.sentence_filter.generate_single_lemmas_list(complex_tag)
+        tags_list = []
+        tags_list.extend(single_tag)
+        tags_list.extend(complex_tag)
+        tags_list.extend(single_tags_from_complex_tags)
+        return tags_list
+
+    def find_best_tags_coverage(self, documents, tags):
+        id_of_best_cov_doc = -1
+        max_conf_from_covered_docs = 0.0
+        for document in documents:
+            tags_from_document = document['tags']
+            coverage = statement_utils.my_intersection(set(tags), set(tags_from_document))
+            confidence, was_accepted = self.is_accepted(coverage, len(set(tags_from_document)), tags)
+            if was_accepted and confidence >= max_conf_from_covered_docs:
+                id_of_best_cov_doc = document['_id']
+                max_conf_from_covered_docs = confidence
+        result_list = list(filter(lambda obj: obj['_id'] == id_of_best_cov_doc, documents))
+        if len(result_list) > 0:
+            return [result_list[0]['text'], max_conf_from_covered_docs]
+        else:
+            return [None, 0.0]
```

Comment:
i think i will be cleaner to use tuples here, if it is possible? you can use tuple similar to list then